orange/src/test/java/com/hartwig/hmftools/orange/algo/purple/PurpleVariantFactoryTest.java

```diff
@@ -119,4 +124,83 @@ public void testCanCreateVariantFromContext()
+                .reportableTranscripts(Set.of("canonical_transcript", "other_transcript1"))
+                .build();
+
+        PaveAlgo paveAlgo = new PaveAlgo(TestEnsemblDataCacheFactory.createDummyCache(), false);
+        PurpleVariant purpleVariant = new PurpleVariantFactory(paveAlgo).fromPurpleVariantContext(context);
+
+        PurpleTranscriptImpact canonicalImpact = purpleVariant.canonicalImpact();
+        assertEquals("canonical_transcript", canonicalImpact.transcript());
+        assertTrue(canonicalImpact.reported());
+
+        List<PurpleTranscriptImpact> purpleOtherImpacts =
+                purpleVariant.otherImpacts()
+                        .stream()
+                        .sorted(Comparator.comparing(PurpleTranscriptImpact::transcript))
+                        .collect(Collectors.toList());
+        assertEquals(2, purpleOtherImpacts.size());
+
+        PurpleTranscriptImpact firstImpact = purpleOtherImpacts.get(0);
+        assertEquals("other_transcript1", firstImpact.transcript());
+        assertTrue(firstImpact.reported());
+
+        PurpleTranscriptImpact secondImpact = purpleOtherImpacts.get(1);
+        assertEquals("other_transcript2", secondImpact.transcript());
+        assertFalse(secondImpact.reported());
+    }
+
+    @Test
+    public void testCanonicalTranscriptNotReported()
+    {
+        ImmutablePurpleVariantContext context = TestPurpleVariantFactory.contextBuilder()
+                .canonicalTranscript("canonical_transcript")
+                .reported(false)
+                .otherImpacts(List.of(transcriptImpact("other_transcript1"), transcriptImpact("other_transcript2")))
+                .reportableTranscripts(Set.of("other_transcript1"))
+                .build();
+
+        PaveAlgo paveAlgo = new PaveAlgo(TestEnsemblDataCacheFactory.createDummyCache(), false);
+        PurpleVariant purpleVariant = new PurpleVariantFactory(paveAlgo).fromPurpleVariantContext(context);
+
+        PurpleTranscriptImpact canonicalImpact = purpleVariant.canonicalImpact();
+        assertEquals("canonical_transcript", canonicalImpact.transcript());
+        assertFalse(canonicalImpact.reported());
+
+        List<PurpleTranscriptImpact> purpleOtherImpacts =
+                purpleVariant.otherImpacts()
+                        .stream()
+                        .sorted(Comparator.comparing(PurpleTranscriptImpact::transcript))
+                        .collect(Collectors.toList());
+        assertEquals(2, purpleOtherImpacts.size());
+
+        PurpleTranscriptImpact firstImpact = purpleOtherImpacts.get(0);
+        assertEquals("other_transcript1", firstImpact.transcript());
+        assertTrue(firstImpact.reported());
+
+        PurpleTranscriptImpact secondImpact = purpleOtherImpacts.get(1);
+        assertEquals("other_transcript2", secondImpact.transcript());
+        assertFalse(secondImpact.reported());
+    }
+
+    private VariantTranscriptImpact transcriptImpact(String transcript)
```

Comment:
Should be annotated with `@NotNull` and should be declared `static`. No kotlin code!!

Reply:
Yeah I totally got out of the habit of thinking of annotations after being in kotlin code for a while!